app/src/main/java/io/github/trojan_gfw/igniter/initializer/InitializerHelper.java

```diff
@@ -16,36 +18,44 @@
  * You should consider carefully to determine which process your initializers are run in.
  */
 public class InitializerHelper {
-    private static List<Initializer> sMainInitializerList;
-    private static List<Initializer> sToolsInitializerList;
+    private static final String TOOL_PROCESS_POSTFIX = ":tools";
+    private static final String PROXY_PROCESS_POSTFIX = ":proxy";
+    private static List<Initializer> sInitializerList;
 
-    static {
-        sMainInitializerList = new LinkedList<>();
-        sToolsInitializerList = new LinkedList<>();
-        registerMainInitializers();
-        registerToolsInitializers();
+    private static void createInitializerList() {
+        sInitializerList = new LinkedList<>();
     }
 
     private static void registerMainInitializers() {
-        sMainInitializerList.add(new MainInitializer());
+        createInitializerList();
```

Comment:
On each invocation of createInitializerList(), it will re-init sInitializerList. Does this correct?

Reply:
It's correct. In fact, one process will only call one of the methods above. For example, method `registerMainInitializers()` will only be called in main process and never be called in tools process or proxy process. Generally, each process runs with its own memory and will not affect others.
So, `sInitializerList` will only be init once in each process. Other process will init `sInitializerList` too, but they are not identically equal.